src/api/libcellml/component.h

```diff
@@ -37,12 +37,19 @@ class LIBCELLML_EXPORT Component: public ComponentEntity, public ImportedEntity
                                   public std::enable_shared_from_this<Component>
 #endif
 {
-public:
+private:
     Component(); /**< Constructor */
+
+public:
     ~Component() override; /**< Destructor */
-    Component(const Component &rhs); /**< Copy constructor */
-    Component(Component &&rhs) noexcept; /**< Move constructor */
-    Component &operator=(Component rhs); /**< Assignment operator */
+    Component(const Component &rhs) = delete; /**< Copy constructor */
+    Component(Component &&rhs) noexcept = delete; /**< Move constructor */
+    Component &operator=(Component rhs) = delete; /**< Assignment operator */
```

Comment:
Similarly (i.e. wrt consistency), it might be nice to have our constructors before the destructor.

Reply:
All classes currently go
* Constructor
* Destructor
* Copy
* Move
* Assignment

If the constructor is not protected or private.  Seems pretty consistent.

Reply:
Ok, my bad, I hadn't realised that. By default, I would have expected all types of constructors to be first and then the destructor. At least, it seems consistent indeed. So, happy with that.